src/r5py/util/memory_footprint.py

```diff
@@ -60,29 +60,64 @@ def share_of_ram(share=0.8, leave_at_least=(2 * (2**10))):
     return share_of_ram
 
 
-def max_memory(max_memory):
-    """Interpret the config parameter --max-memory."""
+def parse_max_memory_string(max_memory):
+    """
+    Extract maximum memory value and unit from text input.
+    """
     try:
-        matches = re.match(r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[%MGT])?", max_memory)
+        matches = re.match(
+            r"(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[%KMGT])?", max_memory
+        )
         value = float(matches["value"])
         unit = matches["unit"]
-        if unit == "%":
-            max_memory = share_of_ram(share=(value / 100.0))
-        else:
-            # convert to MiB
-            if unit is None:
-                value *= 2**-10
-                if value < 1:
-                    value = 1
-            # elif unit == "M":
-            #    value *= 2 ** 1
-            elif unit == "G":
-                value *= 2**10
-            elif unit == "T":
-                value *= 2**20
-            max_memory = round(value)
+
+        if unit is None:
```

Comment:
I’m not really happy to kick out the possibility to specify this in bytes, see my (very easy to implement) suggestion in the relating issue. Use case: e.g. when this value is computed somewhere else (no need to complicatedly convert to +suffix, and back again)

Reply:
Yes, we can do this (see my comment in the issue). However, what is the use-case when someone would really request memory in bytes? 🙂 At least, it would probably make sense to use some other unit as the default, no? Like `MiB`

Reply:
See my comment in the issue, I would find it confusing is the unprefixed unit is anything else but the base unit (bytes). 

But I somehow question the (my) choice of parsing into MiB here. There’s no real reason for it, and in fact it would be weird if we accept bytes, but then pass on a value that’s not bytes (without renaming the constant). The JVM accepts bytes perfectly fine, so why not stay there, in a way. That would also allow to make the if/elif tree much more DRY (the suffix changes the exponent, only).

Reply:
> See my comment in the issue, I would find it confusing is the unprefixed unit is anything else but the base unit (bytes).
> 
> But I somehow question the (my) choice of parsing into MiB here. There’s no real reason for it, and in fact it would be weird if we accept bytes, but then pass on a value that’s not bytes (without renaming the constant). The JVM accepts bytes perfectly fine, so why not stay there, in a way. That would also allow to make the if/elif tree much more DRY (the suffix changes the exponent, only).

Okay, well let's make a separate issue about this change. Will now need to finish with this one.